src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideUnitReportsStep.groovy

```diff
@@ -0,0 +1,219 @@
+            return false
+        }
+        double failed = (results.failCount / results.totalCount) * 100.0
+        return unstableThreshold < failed
+    }
+
+    boolean isFailure(TestResult results) {
+        if (results.totalCount == 0 || failedThreshold <= 0.0) {
+            return false
+        }
+        double failed = results.failCount / results.totalCount * 100.0
+        return failedThreshold < failed
+    }
+
+    protected ArrayList<String> processReport(File reportFile, String reportDirName, String outDirPath, TaskListener listener) {
+        return ZipUtil.extractFilesByGlobPattern(reportFile, DEFAULT_REPORT_GLOB, "${outDirPath}/${reportDirName}")
+    }
+
+    @Override
+    StepExecution start(StepContext context) throws Exception {
+        return new Execution(this, context)
+    }
+
+    static class Execution extends SynchronousNonBlockingStepExecution<Void> {
+        private static final long serialVersionUID = 1L
+
+        private final transient ProvideUnitReportsStep step
+        private final transient Run run
+        private final transient TaskListener listener
+        private final transient Launcher launcher
+
+        Execution(ProvideUnitReportsStep step, StepContext context) {
+            super(context)
+            this.step = step
+            run = context.get(Run.class)
+            listener = context.get(TaskListener.class)
+            launcher = context.get(Launcher.class)
+        }
+
+        ArrayList<String> getUnitReportFilePaths() throws IOException {
+            return launcher.getChannel().call(new AbstractDownloadReportStep.DownloadReportCallable(step, step.publishConfig.timeout, context))
+        }
+
+        TestResult parseReportFiles(ArrayList<String> reportPaths) throws IOException {
+            return launcher.getChannel().call(new ParseReportsExecutable(reportPaths))
+        }
+
+        void addResultsToRun(TestResult result) {
+            TestResultAction action = run.getAction(TestResultAction.class);
+            if (action != null) {
+                action.mergeResult(result, listener);
+            } else {
+                action = new TestResultAction(run, result, listener);
+                run.addAction(action);
+            }
+        }
+
+        @Override
+        protected Void run() throws Exception {
+            try {
+                ArrayList<String> reportPaths = getUnitReportFilePaths()
+                TestResult testResult = parseReportFiles(reportPaths)
+                testResult.tally() // needed, otherwise totalCount is 0 even when it contains test cases
+
+                if (testResult.totalCount == 0) {
+                    listener.logger.println("No unit test results found.")
+                    if(!step.publishConfig.allowMissing) {
+                        throw new Exception("Build result set to ${Result.FAILURE.toString()} due to missing test results. Adjust AllowMissing step property if this is not intended.")
```

Comment:
I guess we are using `AbortExceptions`

Reply:
No, but the Exception is directly catched by the surrounding `try` block